File: src/utils/protocols/iot.ts
import { Protocol } from '../types';

export const IoTProtocols: Protocol[] = [
  {
    name: 'MQTT',
    ports: [1883, 8883],
    description: 'Message Queuing Telemetry Transport',
    scanFunction: async (host: string, port: number) => {
      // MQTT connection logic
      return {
        protocol: 'MQTT',
        version: '3.1.1',
        security: port === 8883 ? 'TLS' : 'None'
      };
    }
  },
  {
    name: 'CoAP',
    ports: [5683, 5684],
    description: 'Constrained Application Protocol',
    scanFunction: async (host: string, port: number) => {
      // CoAP detection logic
      return {
        protocol: 'CoAP',
        version: '1.0',
        security: port === 5684 ? 'DTLS' : 'None'
      };
    }
  }
];

File: src-tauri/src/protocols/iot.rs
```rust
use tokio::net::TcpStream;
use serde::{Serialize, Deserialize};

#[derive(Debug, Serialize, Deserialize)]
pub struct IoTProtocolResult {
    pub protocol: String,
    pub version: Option<String>,
    pub security: IoTSecurity,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct IoTSecurity {
    pub encryption: Option<String>,
    pub authentication: Option<String>,
    pub certificates: bool,
}

pub struct IoTProtocols;

impl IoTProtocols {
    pub async fn detect_mqtt(host: &str, port: u16) -> Result<IoTProtocolResult, String> {
        let is_tls = port == 8883;
        
        Ok(IoTProtocolResult {
            protocol: "MQTT".to_string(),
            version: Some("3.1.1".to_string()),
            security: IoTSecurity {
                encryption: Some(if is_tls { "TLS" } else { "None" }.to_string()),
                authentication: None,
                certificates: is_tls,
            },
        })
    }

    pub async fn detect_coap(host: &str, port: u16) -> Result<IoTProtocolResult, String> {
        let is_dtls = port == 5684;

        Ok(IoTProtocolResult {
            protocol: "CoAP".to_string(),
            version: Some("1.0".to_string()),
            security: IoTSecurity {
                encryption: Some(if is_dtls { "DTLS" } else { "None" }.to_string()),
                authentication: None,
                certificates: is_dtls,
            },
        })
    }
}
```